Maybe this will fix clang thread failures

lindsayad opened the pull request:
Don't use factorization packages which use openmp

I don't trust helgrind with openmp but I also don't trust threading
generally within PETSc solves since PETSc makes little effort at
thread safety. Both superlu_dist and mumps employ openmp so instead
of using those I'm going to default to using the native PETSc LU
factorization (in serial). Only the latter comes out helgrind clean with
respect to not showing any stack traces involving dgemm

lindsayad committed:
Remove some extraneous braces

lindsayad committed:
Don't use factorization packages which use openmp

I don't trust helgrind with openmp but I also don't trust threading
generally within PETSc solves since PETSc makes little effort at
thread safety. Both superlu_dist and mumps employ openmp so instead
of using those I'm going to default to using the native PETSc LU
factorization (in serial)

Refs clang recipe failure on idaholab/moose#30742

lindsayad commented:
1 clang pass

moosebuild commented:
Job [Clang](https://civet.inl.gov/job/2978339/) on f1afe76 : invalidated by @lindsayad

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2978332/), step Docs: sync website on f1afe76 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/30844/site)

This comment will be updated on new commits.

lindsayad commented:
2 clang pass

moosebuild commented:
Job [Clang](https://civet.inl.gov/job/2978339/) on f1afe76 : invalidated by @lindsayad

lindsayad committed:
Shorten phase field AD jacobian test

lindsayad committed:
Adjust default PETSc Jacobian tester tolerances

In practice, 1e-8 is almost never met. The PETSc message itself
talks about being on the order of 10^-8. And difference tolerances
simply don't matter as they are a reflection of the scaling in the
matrix/physics

moosebuild commented:
Job [HPC](https://civet.inl.gov/job/2983215/) on d763d2e : invalidated by @lindsayad

lindsayad commented:
3 clang pass

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2983214/), step Generate coverage on d763d2e wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>a587c9</th>
      <th colspan=4>#30844 d763d2</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.48%</td>
      <td>85.48%</td>
      <td>-0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>114771</td>
      <td>114770</td>
      <td>-1</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>19497</td>
      <td>19498</td>
      <td>+1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/30844/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/30844/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/stochastic_tools)
- [`subchannel`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/subchannel)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/30844/coverage/xfem)

</details>



This comment will be updated on new commits.

loganharbour reviewed (approved):
As you see fit
